mypy/typeanal.py

```diff
@@ -453,6 +433,44 @@ def analyze_callable_type(self, t: UnboundType) -> Type:
         assert isinstance(ret, CallableType)
         return ret.accept(self)
 
+    def analyze_callable_args(self, t: TypeList) -> Optional[Tuple[List[Type],
```

Comment:
Using `t` for the arg name is a little fishy (especially since `t` in the code you refactored out was just a Type, and t here corresponds to t.args there). Maybe `tl`?